### PR TITLE
WIP: Try new publishing flow

### PR DESCRIPTION
**Pull request by sarahmonster**
This is my attempt to start implementing publishing flow changes in #7602. I've done some of the cosmetic changes, but some of the more heavy-lifting will need a friendly developer, especially since some of the code in here I'm finding a bit hard to follow.

I've tried some bigger changes here, with respect to adding a lot more spacing between elements and adding some typographic hierarchy. It may be a bit much here, since it's inconsistent with styling used elsewhere, but we can always dial it back if it feels too disjointed. 

## What's missing?

- [ ] scrolling should only scroll the panel, not the window itself
- [ ] dimming the background via a scrim would be good, and I think this would also solve the interaction issue where the page gets scrolled rather than the panel, by making the page itself non-interactive
- [ ] might make sense to refactor so there's pre-publish-panel and post-publish-panel, since they are now (or will be) more separate entities, and there's a lot of functionality changes happening here—I started doing this but got a bit lost in the weeds, so I'd appreciate some help there
- [x] panel title doesn't seem to be behaving properly, I think because I've moved it up a level and the logic is different
- [ ] there's an awkward lag for suggestion panel to come up (not introduced via this PR but could be nice to get fixed if it's relatively trivial)
- [x] the checkbox for "don't show this again" should probably not show this again (via an option)
<s>- [ ] clicking "private" toggle immediately publishes the post (#9396)</s>
- [ ] changed the password input field to be more consistent with the other inputs and it broke things (will file separately)

## Next phase
I've only dealt with the pre-publish panel right now, mostly due to technical limitations. Here's what I'd like for the post-publish phase:

There should be one state for "publishing" and one for "published" (both of which currently exist), and then, once the post has published, we should redirect to show the post, with a modal overlay giving some next-step advise as per the mockup. 

I can adjust the styling of the "publishing..." and the "published!" states, but it would help to be able to get them to display temporarily, which I'm having troubles making happen. (Setting `submitted: true` on `PostPublishPanel` brings up the published state, but I'm having troubles getting the publishing state to stay on the screen for any length of time, making it super difficult to style or adjust since it disappears quite quickly!)

### CODE REVIEW

File: packages/editor/src/components/post-publish-panel/style.scss [hunk 1]
```diff
@@ -11,68 +21,88 @@
...
 }
 
-.components-button.editor-post-publish-panel__toggle.is-primary {
-	display: inline-flex;
-	align-items: center;
+.editor-post-publish-panel__link {
+	color: $blue-dark-900;
+	font-weight: 400;
+	padding-left: 4px;
+	//text-decoration: underline;
```
Comment (tofumatt) [inline packages/editor/src/components/post-publish-panel/style.scss]: 🔥

File: packages/editor/src/components/post-publish-panel/index.js [hunk 2]
```diff
@@ -97,6 +96,28 @@ class PostPublishPanel extends Component {
...
+					{ submitted && (
+						<div className="editor-post-publish-panel__header-published">
+							{ isScheduled ? __( 'Scheduled' ) : __( 'Published' ) }
+						</div>
+					) }
+				</div>
+
+				<div className="editor-post-publish-panel__disable-check">
+					<CheckboxControl
+						label={ __( 'Don’t show this again' ) }
```
Comment (tofumatt) [inline packages/editor/src/components/post-publish-panel/index.js]: "Don’t show this panel again" might be better; I'm often wary about an unclear "this".

File: packages/editor/src/components/post-publish-panel/index.js [hunk 3]
```diff
@@ -61,22 +61,21 @@ class PostPublishPanel extends Component {
 		this.setState( { loading: true } );
 	}
 
+	onUpdateDisableToggle() {
+		// There should be a setting to allow users to skip the pre-publish step forever.
```
Comment (tofumatt) [inline packages/editor/src/components/post-publish-panel/index.js]: That'll need to be a user setting, I'd think. I don't know how we set those in Gutenberg, but this should dispatch an action that makes an API request to toggle said setting.

**Review by tofumatt — COMMENTED** [review]
Cool. I like the increase in size; the old one felt cramped/small and it didn't get my attention because of that.

Just a few passing thoughts; I know this isn't complete.

**Comment by nosolosw** [comment]
> scrolling should only scroll the panel, not the window itself

It works for me in Chrome and Firefox:

peek 2018-09-06 17-33

**Comment by nosolosw** [comment]
I've got a couple of questions about the "Don't show this again" checkbox that affects implementation

1. If an user checks it, how can that be unchecked if the pre-publish panel is not ever shown?

One potential solution would be to leverage the fact that the pre-publish panel _only_ shows settings that can be modified elsewhere. What if we do the same for this one? I can think of two places for this:

- In the "Status & Visibility" metabox within the Document sidebar.
- In the "Tools" section of the general Gutenberg menu (ellipsis at the top right).

I think it conceptually fits within the "Tools" section. Thoughts?

2. Shall we change the publish button text from "Publish..." to "Publish" depending on whether the pre-publish panel is going to be shown or not?

That'd be more in line with the intent of adding dots to the Publish text (convey that it's more than just publish...), but wanted to check first.

**Comment by sarahmonster** [comment]
For the user setting, I was thinking it'd make most sense to have it under the "Tools" section, since that's where other user settings are. It's pretty similar conceptually to the "Show tips" option that appears here as well. So, basically, your instincts here are bang-on and exactly what I was thinking (but didn't put in the mockup). 💯

> Shall we change the publish button text from "Publish..." to "Publish" depending on whether the pre-publish panel is going to be shown or not?

Yes please!

**Comment by sarahmonster** [comment]
Yep, makes sense! Once that PR is finished I'll rebase this one on master and clean things up a bit, and then we can loop back to this.

**Comment by sarahmonster** [comment]
I've now rebased this to take into account other changes, and it's working a bit better. Here's what's left that I need help with in order for this to be merge-ready!

- add a background scrim to nix interactivity & dim the background
- redirect to post after publishing and show a dismissable post-publish modal
- help with testing out different states for publishing button, so I can customise them in CSS (ie, can I set a state temporarily for testing)

**Comment by gziolo** [comment]
It looks like this one was abandoned, what is the plan moving forward with those changes proposed? Do you plan to refresh this PR? I'm labeling it as `Stale` to make PR triaging easier but feel free to remove it as soon as you refresh PR.

**Comment by sarahmonster** [comment]
I'm going to go ahead and close this. To implement the changes suggested in #7602, I'd need development help—this PR was as far as I could get with it. 

If it's a priority to implement these changes, I'm happy to revisit in a new PR, but I would need help addressing the technical issues outlined above.
